emerge-gitclone

```diff
@@ -37,13 +37,32 @@ def get_channel():
     return channel
 
 
-def git_clone_repo(repo_url, repo_location):
-    """Clone the given repo at the given location"""
+def git_clone_scripts(repo_url, repo_location, ref):
+    """Clone the scripts repo at the given location, and handle submodules (if applicable)"""
     print(f">>> Starting git clone in {repo_location}")
     os.umask(0o022)
     subprocess.check_call(
-        ["git", "clone", "--recurse-submodules", repo_url, repo_location]
+        ["git", "clone", repo_url, repo_location]
     )
+
+    # Check if this ref uses submodules and check these out if it does.
+    subprocess.check_output(
+        ["git", "-C", repo_location, "checkout", ref]
+    )
+    try:
+        with open(repo_location + '/.gitmodules') as gm:
+            gmc = gm.read()
+            if '[submodule ' in gmc:
+                print(f">>> Submodules detected in {ref}.")
+                subprocess.check_output(
+                    ["git", "-C", repo_location, "submodule", "init"]
+                )
+                subprocess.check_output(
+                    ["git", "-C", repo_location, "submodule", "update"]
+                )
+    except (FileNotFoundError, IOError):
+        print(f">>> No submodules detected in {ref}. Assuming unified scripts repo.")
```

Comment:
So there are three cases - no .gitmodules (prints "No submodules detected"), valid .gitmodules (prints "Submodules detected"), and empty or invalid .gitmodules (nothing is printed).

I dunno if the empty .gitmodules can actually happen (I'd treat it as if no .gitmodules file exists), but on the corrupted .gitmodules we probably should just bail out.

Reply:
I think we should print a message and continue, but you're right, we should handle that case.